libraries/AP_LandingGear/AP_LandingGear.h

```diff
@@ -40,6 +41,7 @@ class AP_LandingGear {
 
     /// returns true if the landing gear is deployed
     bool deployed() const { return _deployed; }
+    bool locked() const { return _deploy_lock; }
```

Comment:
This needs a description!  After reading through this patch quickly once I don't get it :-)

In particular, how does this leave the "Unlocked" state and enter the "Retracted" state after the switch is toggled?

Reply:
The gear switch handling in copter switches calls the unlock in the same processes that checks the state and warns the operator.